File: js/array-maximal-adjacent-diff.js
function arrayMaximalAdjacentDifference(inputArray){
    let diffArray = []
    let maxDiff = 0
    
    for(i = 0;i < inputArray.length -1; i++){
        diffArray.push(inputArray[i] - inputArray[i + 1])
        if(diffArray[i] < 0){
            diffArray[i] *= -1
        }
    }
    
    return maxDiff=Math.max(...diffArray)
}

let inputArray = [2, 4, 1, 0]
console.log(arrayMaximalAdjacentDifference(inputArray))

File: go/array-maximal-adjacent-diff/array-maximal-adjacent-diff.go
package main

import(
	"fmt"
)

func arrayMaximalAdjacentDifference(inputArray []int)int{
	res := 0
	diff := 0
	for i := 0; i < len(inputArray) - 1; i++{
		diff = inputArray[i] - inputArray[i + 1]
		if diff < 0 {
			diff *= -1
		}
		if res < diff{
			res = diff
		}
	}
	return res
}

func main(){
	inputArray := []int{2, 4, 1, 0}
	fmt.Printf("The max diff between adjacent elements for the array %v is %d", inputArray, arrayMaximalAdjacentDifference(inputArray))
}